bug(analytics): CodebaseOverviewPanel uses realTimeEnabled prop not declared in Props interface

## Problem

`vue-tsc` reports two `TS2339` errors in `CodebaseOverviewPanel.vue`:

```
src/components/analytics/CodebaseOverviewPanel.vue(9,62): error TS2339: Property 'realTimeEnabled' does not exist on type ...
src/components/analytics/CodebaseOverviewPanel.vue(11,18): error TS2339: Property 'realTimeEnabled' does not exist on type ...
```

## Root Cause

The template (lines 9 and 11) references `realTimeEnabled` in two places:
```html
<div class="refresh-indicator" :class="{ active: realTimeEnabled }">
  {{ realTimeEnabled ? $t('...live') : $t('...static') }}
```

But the `Props` interface only declares:
```ts
interface Props {
  systemOverview: SystemOverviewData | null
  communicationPatterns: CommunicationPatternsData | null
  codeQuality: CodeQualityData | null
  performanceMetrics: PerformanceMetricsData | null
}
```

`realTimeEnabled` is missing.

## Fix

Add `realTimeEnabled: boolean` to the `Props` interface (with a sensible default via `withDefaults`).

## Acceptance Criteria
- [ ] `vue-tsc --noEmit -p tsconfig.app.json` reports no TS2339 errors in `CodebaseOverviewPanel.vue`
- [ ] Live/static indicator renders correctly based on prop value

--- a/autobot-frontend/src/components/analytics/CodebaseOverviewPanel.vue
+++ b/autobot-frontend/src/components/analytics/CodebaseOverviewPanel.vue
@@ -199,6 +199,7 @@ interface Props {
   communicationPatterns: CommunicationPatternsData | null
   codeQuality: CodeQualityData | null
   performanceMetrics: PerformanceMetricsData | null
+  realTimeEnabled?: boolean
 }
 
 defineProps<Props>()

